addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -212,23 +212,26 @@ def _peppol_get_new_documents(self, skip_no_journal=False):
                     document_content = content["document"]
                     filename = content["filename"] or 'attachment'  # default to attachment, which should not usually happen
                     decoded_document = edi_user._decrypt_data(document_content, enc_key)
-                    attachment = self.env["ir.attachment"].create(
-                        {
-                            "name": f"{filename}.xml",
-                            "raw": decoded_document,
-                            "type": "binary",
-                            "mimetype": "application/xml",
-                        }
-                    )
-                    created_moves |= edi_user._peppol_import_invoice(attachment, None, content["state"], uuid, journal)
+                    attachment = self.env["ir.attachment"].create({
+                        "name": f"{filename}.xml",
+                        "raw": decoded_document,
+                        "type": "binary",
+                        "mimetype": "application/xml",
+                    })
+                    vals_to_ack = edi_user._peppol_import_invoice(attachment, content["state"], uuid, journal=journal)
+                    if move_to_ack := vals_to_ack.get('move'):
+                        created_moves |= move_to_ack
+                    if uuid_to_ack := vals_to_ack.get('uuid'):
+                        uuids_to_ack.append(uuid_to_ack)
```

Comment:
More a note for a future task. We could heavily speedup that part by batching the creation of the attachments and the creation of invoices